runtime/interpreter/statictype.go

```diff
@@ -605,8 +613,6 @@ func (e EntitlementMapAuthorization) Equal(auth Authorization) bool {
 // ReferenceStaticType
 
 type ReferenceStaticType struct {
-	// BorrowedType is the type of the usage (T in &T)
-	BorrowedType StaticType
 	// ReferencedType is type of the referenced value (the type of the target)
 	ReferencedType StaticType
```

Comment:
Why was the borrowed type removed? Isn't it still needed?

Reply:
No, the "BorrowedType" was previously used to check what the "visible" type of the reference was, for things like casting where the type behavior of the reference was not necessarily the same as the type behavior of its underlying type. This has been simplified, a reference's borrow type is the same as the type of its underlying value at runtime (i.e. downcasting is possible), so we don't need this anymore.

Reply:
I see, that makes sense. My confusion probably stemmed from the fact that I assumed we would keep `BorrowedType`, because that was the actual "static type", and remove `ReferencedType` as that was the "dynamic type". 

It would be good to keep the comment though: `ReferencedType is the type T in &T`. The current comment, "referenced value (the type of the target)" meant that it was actually the concrete type of the value that the reference value was pointing to. That is not the case anymore (I assume?)

Reply:
Could we remove `ReferencedType` instead of `BorrowedType`? I have a bad feeling we currently use `BorrowedType` somewhere, initialize `ReferencedType` to a different value, and then the PR switches the use to `ReferencedType` and suddenly the semantics are wrong. It would reduce the number of changes in the PR

Reply:
The comment as written is accurate; `ReferencedType` is the type of the referenced value. The type of the `T` that appears in the `&T` is no longer part of the static type, as it is not necessary, as we can now make all up and downcasting decisions based on the runtime type of the referenced value, and do not need to consider the type it was originally borrowed with. The exception to this are the entitlements of the reference, which are tracked separately. No other static types keep a converted copy of the original sema type around; references only did this because previously the runtime value was only allowed to be used in ways that reflected its original borrow type. 

The only places in the code where this actually comes up are in the interpreter encoding (which used to encode the `BorrowedType` and now uses the `ReferencedType`), in `IsSubtypeOfSemaType` in the interpreter (which was changed before this field was removed, and was the impetus for removing it in the first place), and in the creation of sema types from static types for references (which is also intentional).

Reply:
Thank you for the great explanation! 👍